contributors/design-proposals/statefulset-update.md

```diff
@@ -0,0 +1,847 @@
+is the case when a user wants to perform a canary update or a phased roll out, 
+they are partitioned around an ordinal such that all Pods with a lower ordinal 
+conform to one PodTemplate and all Pods with a greater or equal ordinal 
+conform to the other. The conditions that define this state in terms of the 
+StatefulSet's StatefulSetSpec and StatefulSetStatus are below.
+
+1. The StatefulSet contains exactly `[0,.Spec.Replicas)` Pods.
+1. If StatefulSet's `.Spec.GenerationPartition` is nil, then the following is 
+true.
+    1. The StatefulSet's `.Status.CurrentTemplateGeneration` is equal to its 
+    `.Status.UpdatedTemplateGeneration`.
+    1. All Pods in the StatefulSet have been generated from the PodTemplate 
+    labeled with a `StatefulSetTemplateGenerationLabel` equal to its 
+    `.Status.TemplateGeneration`.
+1. If the StatefulSet's `.Spec.GenerationPartition` is not nil, then the 
+following is true.
+    1. All Pods with ordinals is the sequence `[0,.Spec.GenerationPartition)` have 
+     been generated from the PodTemplate in the StatefulSet's revision history
+     that is labeled with a `StatefulSetTemplateGenerationLabel` equal to 
+     `.Status.TemplateGeneration`.
+    1. All Pods with ordinals in the sequence 
+     `[Spec.GenerationParition,.Spec.Replicas)` have been created with the 
+     PodTemplate in the StatefulSet's revision history that is labeled with a 
+     `StatefulSetTemplateGenerationLabel` equal to 
+     `.Status.UpdatedTemplateGeneration`.
+
+### Revised Controller Algorithm
+The StatefulSet controller will use the following algorithm to continue to 
+make progress toward the user declared [target state](#target-state) while 
+respecting the controller's 
+[identity](https://kubernetes.io/docs/concepts/abstractions/controllers/statefulsets/#pod-identity), 
+[deployment, and scaling](https://kubernetes.io/docs/concepts/abstractions/controllers/statefulsets/#deployment-and-scaling-guarantee) 
+guarantees.
+
+1. The controller will 
+[reconstruct the revision history](#history-reconstruction) of the StatefulSet.
+1. The controller will process any [template updates](#template-updates) to 
+ensure that the StatefulSet's revision history is consistent with the user 
+declared desired state.
+1. The controller will select all Pods in the StatefulSet, filter any Pods not 
+owned by the StatefulSet, and sort the remaining Pods in ordinal order.
+1. If any Pods with ordinals in the sequence `[0,.Spec.Replicas)` have not been 
+created, for the Pod corresponding to the lowest such ordinal, the controller 
+will [select the PodTemplate](#podtemplate-selection) from the StatefulSet's 
+revision history corresponding to the Pod's ordinal and create the Pod.
+1. If all Pods in the sequence `[0,.Spec.Replicas)` have been created, but if any 
+have a do not have a Ready Condition, the StatefulSet controller will wait for 
+these Pods to either become Ready, or to be completely deleted. 
```

Comment:
If a Pod was not able to mount its volume successfully because the volume could not be provisioned successfully  or some other reason, does that count as Pod not ready ?